ql/src/java/org/apache/hadoop/hive/ql/ddl/table/info/desc/formatter/TextDescTableFormatter.java

```diff
@@ -129,7 +168,7 @@ private void addPartitionTransformData(DataOutputStream out, Table table, boolea
     out.write(partitionTransformOutput.getBytes(StandardCharsets.UTF_8));
   }
 
-  private void addStatsData(DataOutputStream out, String columnPath, List<FieldSchema> columns, boolean isFormatted,
+  private void addStatsData(DataOutputStream out, String columnPath, Table table, List<FieldSchema> columns, boolean isFormatted,
```

Comment:
I think it would possibly be beneficial to put the constraint info into the `FieldSchema` itself in some usable way ...not sure what would be the earliest point we could set it but it could make constraint info more accessible...

maybe return with a non-thrift `FieldSchema` from the `Table` which is constriant aware?

we are always making circles because data is not where it supposed to be...